packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -1052,15 +1052,23 @@ class _AppBarState extends State<AppBar> {
         ?? appBarTheme.systemOverlayStyle
         ?? _systemOverlayStyleForBrightness(ThemeData.estimateBrightnessForColor(backgroundColor));
 
+    final double elevation = widget.elevation
+      ?? appBarTheme.elevation
+      ?? _defaultElevation;
+    // Sets MaterialType to 'transparency' if background color is transparent
+    // and elevation is zero.
+    final MaterialType materialType = backgroundColor == const Color(0x00000000) && elevation == 0.0
```

Comment:
what if the color is something like transparent blue (Colors.blue.withOpacity(0.0) for example)? does flutter/dart:ui already convert it to transparent black or would it be the actual color with 0 alpha/opacity? because if it's the latter then this check would potentially fail